File: tests/fixtures/icu_admission.c
#include <stdbool.h>

/*
 * ICU 入住資格評估（C 版）k=10
 * 對應 Python 版 icu_admission.py
 */
bool check_icu_admission(int age, bool low_bp, bool high_heart_rate, bool high_resp_rate,
                         bool high_temp, bool low_gcs, bool low_oxygen, bool low_urine,
                         bool high_creatinine, bool sepsis) {
    if (
        age >= 18
        && (low_bp || high_heart_rate)
        && (high_resp_rate || high_temp)
        && (low_gcs || low_oxygen)
        && (low_urine || high_creatinine || sepsis)
    ) {
        return true;
    }
    return false;
}

File: fixtures/icu_admission.py
"""
ICU 入住資格評估（k=10）
使用布林參數避免 Z3 型別衝突
"""


def check_icu_admission(
    age: int,
    low_bp: bool,
    high_heart_rate: bool,
    high_resp_rate: bool,
    high_temp: bool,
    low_gcs: bool,
    low_oxygen: bool,
    low_urine: bool,
    high_creatinine: bool,
    sepsis: bool,
) -> bool:
    """ICU 入住資格評估（k=10）"""
    if (
        age >= 18
        and (low_bp or high_heart_rate)
        and (high_resp_rate or high_temp)
        and (low_gcs or low_oxygen)
        and (low_urine or high_creatinine or sepsis)
    ):
        return True
    return False
